### PR TITLE
DB-9048 Support all types for create/alter table with default 

### PR DESCRIPTION
**Pull request by arnaud-splice**

### CODE REVIEW

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: db-engine/src/main/java/com/splicemachine/db/iapi/types/DataTypeDescriptor.java [hunk 1]
```diff
@@ -1756,6 +1756,10 @@ public void setChildren(DataTypeDescriptor[] children) {
      }
 
     public DataValueDescriptor getDefault() throws StandardException {
+        if (typeId.getTypeFormatId() == StoredFormatIds.BIT_TYPE_ID) {
+            return new SQLBit(new byte[typeDescriptor.getMaximumWidth()]);
```
Comment (dgomezferro) [inline db-engine/src/main/java/com/splicemachine/db/iapi/types/DataTypeDescriptor.java:1760]: Why do we need to create it with maximumWidth instead of 0?

Comment (arnaud-splice) [reply db-engine/src/main/java/com/splicemachine/db/iapi/types/DataTypeDescriptor.java:1760]: We mimic what DB2 is doing. For char(n) for bit data, their default is a byte array of size n, filled with 0x0

**Pull request merged**